characteristic_secure/web_app.js

```diff
@@ -0,0 +1,179 @@
+        BluetoothUUID.getCharacteristic(ageCharacteristicName));
+      assertEquals(characteristic.service, service,
+        'characteristic service mismatch');
+      assertEquals(ageCharacteristicID, characteristic.uuid,
+        'incorrect characteristic UUID');
+
+      let dataView = await characteristic.readValue();
+      let val = dataView.getUint8(0, /*littleEndian=*/true);
+      logInfo(`Starting age is ${val}`);
+
+      logInfo(`Setting age to 25`);
+      await characteristic.writeValueWithResponse(new Uint8Array([25]));
+
+      dataView = await characteristic.readValue();
+      val = dataView.getUint8(0, /*littleEndian=*/true);
+      assertEquals(25, val, 'incorrect age');
+      logInfo('readback success.');
+
+      logInfo(`Setting to 73`);
+      await characteristic.writeValueWithResponse(new Uint8Array([73]));
+
+      dataView = await characteristic.readValue();
+      val = dataView.getUint8(0, /*littleEndian=*/true);
+      assertEquals(73, val, 'incorrect age');
+      logInfo('readback success.');
+    }
+
+    const testHeartRateService = async (gettServer) => {
+      logInfo(`Requesting service: "${heartRateServiceName}"...`);
+      const service = await gattServer.getPrimaryService(
+        BluetoothUUID.getService(heartRateServiceName));
+      assertEquals(device, service.device, 'service device mismatch');
+      assertEquals(heartRateServiceID, service.uuid, 'incorrect service UUID');
+
+      logInfo(`Connected to service uuid:${service.uuid}, primary:${service.isPrimary}`);
+      logInfo(`Requesting characteristic "${heartRateMaxCharacteristicName}"...`);
+      const characteristic = await service.getCharacteristic(
+        BluetoothUUID.getCharacteristic(heartRateMaxCharacteristicName));
+      assertEquals(characteristic.service, service,
+        'characteristic service mismatch');
+      assertEquals(heartRateMaxCharacteristicID, characteristic.uuid,
+        'incorrect characteristic UUID');
+
+      let dataView = await characteristic.readValue();
+      let val = dataView.getUint8(0, /*littleEndian=*/true);
+      logInfo(`Starting max hart rate is ${val}`);
+
+      logInfo(`Setting maximum heart rate to 67`);
+      await characteristic.writeValueWithResponse(new Uint8Array([67]));
+
+      dataView = await characteristic.readValue();
+      val = dataView.getUint8(0, /*littleEndian=*/true);
+      assertEquals(67, val, 'incorrect maximum heart rate');
+      logInfo('readback success.');
+
+      logInfo(`Setting to 110`);
+      await characteristic.writeValueWithResponse(new Uint8Array([110]));
+
+      dataView = await characteristic.readValue();
+      val = dataView.getUint8(0, /*littleEndian=*/true);
+      assertEquals(110, val, 'incorrect maximum heart rate');
+      logInfo('readback success.');
+    }
+
+    const options = {
+      filters: [{ services: [nordicUARTService] }],
+      optionalServices: [ageServiceNumber,
+        BluetoothUUID.getService(heartRateServiceName)]
```

Comment:
I appreciate the desire to hit 100% function coverage in this test suite but these functions are already well tested by the fully automated Web Platform Tests and using them here adds confusion to an already complex test case.

Reply:
Done.